### PR TITLE
Permission requirements

### PR DESCRIPTION
**Pull request by rabbitstack**
127 0 0 1_8000_agents_sematext-agent_permission-requirements_

### CODE REVIEW

File: docs/agents/sematext-agent/permission-requirements.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+- load eBPF probes on kernel symbols
+- monitoring network traffic
+- accessing process log files
+- opening the `/dev/kmsg` device for reading kernel log buffer messages
+
+## Container setups
+
+In container setups, Sematext Agent requires that the following bind mounts are defined. Please note, the majority of bind mounts are propagated to container as read-only volumes.
+
+- `/:/hostfs:ro` - the root partition is mounted in order to collect disk space metrics from inside container.
```
Comment (adnanrahic) [inline docs/agents/sematext-agent/permission-requirements.md:16]: space metrics from inside container
=>
space metrics from inside the container

File: docs/agents/sematext-agent/permission-requirements.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- `/sys/:/hostfs/sys:ro` - sysfs is needed for scanning the cgroup hierarchies for container metrics collection.
+-  `/var/run/:/var/run/` - Sematext Agent has a persistent journal where it writes events that failed to ship. By default, the agent stores events in `/var/run/st-agent` directory on the host. This means that even if the container is removed, the journal is still accessible for future agent containers.
```
Comment (adnanrahic) [inline docs/agents/sematext-agent/permission-requirements.md:18]: agent stores events in
=>
agent stores events in the

File: docs/agents/sematext-agent/permission-requirements.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+- `/sys/kernel/debug:/sys/kernel/debug` - is required by eBPF engine to attach kprobes on kernel symbols. Note, if your system doesn't support eBPF, this bind mount is not required.
```
Comment (adnanrahic) [inline docs/agents/sematext-agent/permission-requirements.md:19]: required by eBPF engine
=>
required by the eBPF engine

File: docs/agents/sematext-agent/permission-requirements.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- `/etc/passwd:/etc/passwd:ro` and `/etc/group:/etc/group:ro` - for the purpose of resolving user/group identifiers to their symbolic names agent needs to access these files. Users/groups are collected as part of process metadata.
```
Comment (adnanrahic) [inline docs/agents/sematext-agent/permission-requirements.md:20]: identifiers to their symbolic names agent needs
=>
identifiers to their symbolic names, the agent needs

File: docs/agents/sematext-agent/containers/installation.md [hunk 5]
```diff
@@ -91,3 +93,5 @@ docker service create --mode global --name st-agent \
 -e REGION=<US or EU> \
 sematext/agent:latest
 ```
+
+_[Read](../permission-requirements.md#container-setups) more about why Sematext Agent needs access to host' files and directories._
```
Comment (adnanrahic) [inline docs/agents/sematext-agent/containers/installation.md:97]: _[Read](../permission-requirements.md#container-setups) more about why Sematext Agent needs access to host' files and directories._

=>

_[Read more](../permission-requirements.md#container-setups) about why Sematext Agent needs access to host files and directories._

File: docs/agents/sematext-agent/kubernetes/installation.md [hunk 6]
```diff
@@ -240,6 +240,8 @@ spec:
         key: node-role.kubernetes.io/master
 ```
 
+_[Read](../permission-requirements.md#container-setups) more about why Sematext Agent needs access to host' files and directories._
```
Comment (adnanrahic) [inline docs/agents/sematext-agent/kubernetes/installation.md:243]: _[Read](../permission-requirements.md#container-setups) more about why Sematext Agent needs access to host' files and directories._

=>

_[Read more](../permission-requirements.md#container-setups) about why Sematext Agent needs access to host files and directories._

**Review by adnanrahic — CHANGES_REQUESTED** [review]

**Comment by otisg** [comment]
*Could* one run the agent without root permission?  If so, what would (not) happen?

**Comment by rabbitstack** [comment]
Good point. I'll write something about rootless permission.

**Review by adnanrahic — APPROVED** [review]

**Pull request merged**